vortex-file/src/strategy.rs

```diff
@@ -113,7 +129,90 @@ impl LayoutWriter for BtrBlocksCompressedWriter {
         // Compute the stats for the chunk prior to compression
         chunk.statistics().compute_all(STATS_TO_WRITE)?;
 
-        let compressed = BtrBlocksCompressor.compress(&chunk)?;
+        let mut compressed_array = None;
+
+        // If we have information about the previous chunk
+        if let Some(prev_compression) = self.previous_chunk.as_ref() {
+            let prev_chunk = prev_compression.chunk.clone();
+            let prev_vtable = prev_chunk.vtable();
+            let canonical = chunk.to_canonical()?;
+
+            log::debug!(
+                "Trying to encode {} array into {}",
+                canonical.as_ref().encoding(),
+                prev_vtable.id()
+            );
+            match prev_vtable.encode(&canonical, Some(&prev_chunk))? {
+                // Encoding isn't supported, so we remove the previous chunk state to let the compressor to try again.
+                None => {
+                    self.previous_chunk.take();
+                }
+                Some(encoded) => {
+                    let prev_children = prev_compression.chunk.named_children();
+                    let encoded_children_names = encoded.named_children();
+
+                    let mut new_map = HashMap::new();
+                    for (child_name, child) in encoded_children_names.into_iter() {
+                        // If there's a matching child, we try and encode the child
+                        if let Some((_, prev_child)) =
+                            prev_children.iter().find(|(name, _)| name == &child_name)
+                        {
+                            if let Some(new_encoded_child) = prev_child
+                                .vtable()
+                                .encode(&child.to_canonical()?, Some(prev_child))?
+                            {
+                                new_map.insert(child_name.clone(), new_encoded_child);
+                            } else if prev_child.encoding() != child.encoding() {
+                                log::warn!(
+                                    "Couldn't encode {} array as {}",
+                                    child.encoding(),
+                                    prev_child.encoding()
+                                )
+                            }
+                        }
+
+                        // If we didn't encode the child, we keep the existing one.
+                        if !new_map.contains_key(&child_name) {
+                            new_map.insert(child_name, child);
+                        }
+                    }
```

Comment:
Don't quite get why we match on name and not on position

Reply:
The two chunks might have different number of children, for example when one has validity and the other doesn't. Using names helps keep the number consistent and match between both arrays.

Reply:
Still using names, I got rid of the additional allocated hashmaps, which I think is faster + just less alocations and code.